### DISCUSSION TITLE
How can I authorize user in edge function?

### DISCUSSION THREAD

**Discussion by damiankomonski** [supabase/supabase#21907] [Questions] [answered]
Hello, I have edge function written in Express.js . I would like to ask how can I make the endpoint only available to logged in users? What should I use?

Thanks

**Comment by GaryAustin1** [comment, marked as answer]
You can pass in their jwt token and decode it with a jwt library to make sure it is valid (use the jwt secret from the dashboard).  Or you could call getUser with it to have Supabase check the jwt and return the user data.

**Reply by damiankomonski** [reply]
How can I pass JWT Token?

**Reply by GaryAustin1** [reply]
That is sort of up to you and your app I would think.   You could use an Authorization header (or any custom header) or a parameter.

**Reply by damiankomonski** [reply]
Ok but how can I get JWT Token on the front end? As I understand I have to pass it in fetch request and check it on server?

**Reply by GaryAustin1** [reply]
It is in your signed in user's session info (the access token).

**Reply by baloodevil** [reply]
Getting a signed jwt token is a large topic.  I use Oauth social providers like Google and Facebook.  I suggest looking at [Supabase Auth documentation](https://supabase.com/docs/guides/auth).

**Reply by damiankomonski** [reply]
Hm ok I am thinking how can I decode it on the server. You wrote about JWT Secret from dashboard. How can I do this?

**Reply by GaryAustin1** [reply]
With a jwt library that works with node.js.  I don't use it but there should be one or more options.

Example library:  https://www.npmjs.com/package/jsonwebtoken

**Reply by damiankomonski** [reply]
`const decoded = jwt.verify(token, 'shhhhh');`

**Reply by damiankomonski** [reply]
Ok thanks I will try

**Reply by baloodevil** [reply]
Ah, I see what you are asking.  Good luck with it!

**Comment by damiankomonski** [comment]
One question yet if I can. I am running edge function locally. I create client in this way:
```
  const supabase = createClient(
    Deno.env.get("SUPA_URL"), 
    Deno.env.get("SUPA_ANON_KEY"), 
    {
      global: {
        headers: { Authorization: req.get('Authorization')! },
      },
    }
  );
```

Should I pass local anon_key or anon_key from supabase (production)? Also URL? 

In my app I have client from production. I dont connect with my local database, auth etc.
